### ISSUE TITLE
i18n issues

### ISSUE THREAD

**Issue by paglias** [HabitRPG/habitica#3168]
Issues regarding i18n.habitrpg.com
- [x] private repo
- [ ] https://github.com/HabitRPG/habitrpg/issues/3168#issuecomment-39740879
- [ ] latin american spanish and english uk
- [ ] static pages load only static strings (same for app)
- [x] pet names
- [x] api messages (habitrpg-shared)
- [ ] api messages (server -> controllers: auth, user, groups, payments, challenges)
- [x] attributes and class names https://dl.dropboxusercontent.com/u/105846917/Habit/011.png https://dl.dropboxusercontent.com/u/105846917/Habit/010.png and https://dl.dropboxusercontent.com/u/105846917/Habit/011.png

**Comment by paglias** [comment]
[EDITED to keep track of what has been done]
by @annesoares

> //--- Assinatura / Comprando gemas - Subscription / Buying Gems
> "Payment Methods:" is not translated.

:+1: 

> //--- Equipamentos - Gear
> Class names are still in english.

:+1: 

> Tutorial:
> "aprendiz de wizard"- "wizard" is not translated.
> There's no way to distribute the attribute points. I also have none.
> Also, the attribute names are still in English.

**Comment by deilann** [comment]
> Will look at those, I'm not sure we can translate next prev and end tour but I suppose it isn't a big problem

Wouldn't it just be a better idea to have simply forward and backward arrows and a big X?

**Comment by paglias** [comment]
@deilann ok it looks like we can override the basic template http://bootstraptour.com/api/, I'm just not sure if it is a 0.9.x feature or it is avalaible also in 0.8.x (the one we're using) and if something has to be changed in order to updgrade to 0.9.x

**Comment by deilann** [comment]
[Looks like it's been available since v0.5.0](https://github.com/sorich87/bootstrap-tour/releases)

Do you think it's better to just have < > x or strings?

**Comment by paglias** [comment]
@deilann it's the same for me, maybe strings are used widely

**Comment by deilann** [comment]
I'm leaning towards the symbols because some languages might have awkwardly long versions which could look weird, and no good abbreviations?

**Comment by paglias** [comment]
ok, I'm fine with it

2014-03-30 13:58 GMT+02:00, deilann notifications@github.com:

> I'm leaning towards the symbols because some languages might have awkwardly
> long versions which could look weird, and no good abbreviations?
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/HabitRPG/habitrpg/issues/3168#issuecomment-39023737

## 

Matteo Pagliazzi - paglias.net

**Comment by deilann** [comment]
Added to the checklist!

**Comment by annesoares** [comment]
I sent the same message to you and Marcos Correa (portuguese reviewer), so the strings that are translated incorrectly he probably fixed them ;)

**Comment by MaskedChimp** [comment]
I just finished fixing translations that Anne pointed out. What is not translated is because it's not at Transifex to be translated.

@paglias About "Party Name", the problem is that "Party" and "Name" are separated in transifex. I can switch the translation of those two, but I don't know if you prefer that way.

**Comment by paglias** [comment]
@MarcosHK about "Party Name" it's fixed now

**Comment by paglias** [comment]
@deilann it looks like this:

![selezione_051](https://cloud.githubusercontent.com/assets/589911/2606200/2aa3e2d4-bb4d-11e3-837c-4e49f76973d9.png)

And we have to keep "End tour" as a string, an "x" wouldn't be enough I think

**Comment by ghost** [comment]
@paglias 
If I get it right, we may point out the bugs in translation implementation in this thread?
If so, spacing between "contribText" and "readMore" in Stats and Achievement is lost. I'll open an issue on transifex too.

**Comment by paglias** [comment]
@GitHubSphinx thanks, fixed

**Comment by ghost** [comment]
https://www.transifex.com/projects/p/habitrpg/translate/#ru/frontjson/23804891

> conquor - typo for conquer in the original

https://www.transifex.com/projects/p/habitrpg/translate/#ru/frontjson/23804894

> "accomodates" should be "accommodates"

https://www.transifex.com/projects/p/habitrpg/translate/#ru/frontjson/23804897

> eachother - space missing in the original
> "2" - typo for "?"

https://www.transifex.com/projects/p/habitrpg/translate/#ru/frontjson/23804904 and
https://www.transifex.com/projects/p/habitrpg/translate/#ru/frontjson/23804892
Full stop at the end?

**Comment by ghost** [comment]
@paglias 
As agreed, I remind you about the lines that need little differentiation.
https://github.com/HabitRPG/habitrpg/issues/2932 and https://github.com/HabitRPG/habitrpg/pull/2966

> 1 (crucial). Different versions of Gems when it is used as a title and when it expresses a quantity.
> It would be even better if we could reorder like "Gems: %number%" in the latter case.
> 2 (very much wanted). Participants is used both (a) on the Challenges page (e.g. 38 Participants) and (b) in the quest block on the team page (Participants: user1, user2 etc.). In order to be grammatically correct these should be two different lines: a) "%number_of% Participants" and b) "Participants".
> 3 (just a suggestion). User profile settings have a header "Blurb" and a grayed comment in the field below: "Blurb". Can we make 2 separate lines: e.g. "Blurb" and "Introduce yourself" or "About myself" - to make it more clear in languages that do not have an exact equivalent for blurb.

Comments on 1 and 2 mostly by @oleg-sh.

**Comment by ghost** [comment]
~~For the testing purposes can we get some variables altered on i18n?
1. The health of bosses (make them weaker so that they can be defeated in some minutes)
2. Daily drop caps (better to have infinite to be able to complete collecting quests in some minutes)~~

**Comment by paglias** [comment]
Classes and attributes now use translated version

**Comment by ghost** [comment]
@paglias 
I keep getting error messages at i18n, when it comes to in-game items. When you buy something for gold, when you getting a drop etc. - an error appears saying "Please reload, "Internal Server Error" and there is no effect of actions.
Console gives the following:

> Bought leather helmet  app-bac5371c.js:22
> POST http://i18n.habitrpg.com/api/v2/user/batch-update?_v=62&data=1397147376241&siteVersion=1 500 (Internal Server Error) app-bac5371c.js:5
> Object {data: "Internal Server Error", status: 500, headers: function, config: Object, statusText: "Internal Server Error"}
>  app-bac5371c.js:22

Is it meant this way? I'm not available to complete collecting quests...

BTW confirmation will be needed as it may be something on my side...

**Comment by paglias** [comment]
@GitHubSphinx thanks for pointing this out. It's not a problem on your side, I hope to solve it today or tomorrow

**Comment by ghost** [comment]
@paglias 
Translation is missing: "Or" on popup login window 
http://oi62.tinypic.com/2vngmkm.jpg

**Comment by ghost** [comment]
Just a thought:
does "static pages load only static strings (same for app)" refer also to the question of user not being able to change the language of static page (it's chosen according to system default one)?

**Comment by paglias** [comment]
@lefnire Oook I think it's ready for beta, the branches are names `paglias/translate` here, on habitrpg-shared and on habitrpg-shared private (where I didn't run `grunt`). But only for beta, otherwise you can try it on i18n.habitrpg.com

**Comment by paglias** [comment]
@lefnire everything is rebased to the last commit on develop, let me know if you want to have all the commits into a single one

**Comment by paglias** [comment]
@GitHubSphinx or fixed, errors on i18n.habitrpg.com should be gone.

> Just a thought:
> does "static pages load only static strings (same for app)" refer also to the question of user not being able to change the language of static page (it's chosen according to system default one)?

No.. it means that only the translations necessary for those pages are loaded, we may add a select in order to allow users to change the language on static pages but I'm not sure it would be very useful

**Comment by ghost** [comment]
@paglias I believe it really would. Not many people would notice that, but those who use devices with different system languages e.g. on vacations or business trips we'll surely appreciate that. Have you ever come across a message (popup or just somewhere on top of the page) like "we've chosen English for you, if it's not your language, please select another one:..." Static pages are the cover by which newcomers judge habitrpg. We may want to show some extra care. And if such choice is made it should be applied to in-game interface too.

Added: As for i18n, there's still a problem with drops - as it seems, all other than food, and Egg hunt drops too.

**Comment by MaskedChimp** [comment]
@paglias When you try to sell an item at the market, the name of the item is not showing up entirely. For example, Red Potion, the button shows up as "Sell Red for 3 Gold".

**Comment by ghost** [comment]
@paglias Could you please consider the following (from Tavern chat): 
I suggested:

> there might also be a notice (by Daniel?) that some foreign-speaking guilds can be found in the guilds section. If everything we've translated would be made live non-English speaking players might perhaps be confused a bit by the Tavern chat.

Alys repled:

> I think it would be worth having a separate list somewhere of all the non-English guilds because I suspect it's not easy to find them by scrolling through the full guild list, and perhaps you can't rely on searching for a language name because of different ways a language name can be written ("Belgi(que/ë/en/um)")

**Comment by deilann** [comment]
newbies guild was also requested as a link

**Comment by ghost** [comment]
@deilann yes, but that is not the question of i18n, I believe. So I thought a separate ticket or trello thing may be needed for that

**Comment by deilann** [comment]
That's not really an issue of i18n either; would require a list of guilds
accessible to all users, not localized -- would make sense to have major
language guilds and newbie's guild listed together.

**Comment by paglias** [comment]
@GitHubSphinx @deilann is right this isn't an issue directly related to i18n and not something I have time to work on right now.

As of i18n I think it can go live at least on beta, I've been on vacation for a few days and I think there have been some changes in the code that need to be addressed

**Comment by paglias** [comment]
@lefnire can we merge this? the PRs are #3010 and HabitRPG/habitrpg-shared#168 only to beta of course

**Comment by ghost** [comment]
@paglias 
Concerning the Egg hunt:
@deilann cleared it our for me about the mechanics of naming items in Egg hunt (thanks again!)
So could we please have "Egg-as-a-pet" an independent variable from "Egg" so that we could compile a phrase "Egg-as-a-pet"+"Egg" and which is more important "Potion name"+"Egg-as-a-pet" - just for the sake of grammar, with no quest spoilers or whatever. Please! ))

BTW, it'll be great if we could see the translation on beta.

**Comment by paglias** [comment]
@GitHubSphinx thanks for the help on trello! About egg-as-a-pet, could you make me an example of how it would appear in english?

About beta... I know I know, it's a big change, which changes how parts of the site are written and I can't merge it on my own... the other guys are very busy right now

**Comment by ghost** [comment]
Oops. Seems like I've deleted the post with apologies for messing it up.
@paglias 
I understood that I got it wrong as I thought that "Egg Egg" was compiled from questEggEggText+questEggEggText. But now I see it's rather mady be questEggEggText*egg.
If it's really like that than I'm so embarrassed for taking you time, actually both of you, @paglias and @deilann.

Now at least I've figured out completely how to translate the thing! )

And I didn't mean to hurry you up with beta, just wanted to support you call to @lefnire !

**Comment by deilann** [comment]
Matteo, Sphinx wants a string that's separate from the egg we use for all
eggs to name the egg pet. So a string like "wolf" but for eggs.

**Comment by ghost** [comment]
@deilann now that I've looked closer, I think it's already implemented... Sorry!

**Comment by deilann** [comment]
:thumbsup:

**Comment by ghost** [comment]
@paglias wow thanks for the beta release! The two questions are:
1. how much time do we have for testing before going live completely?
2. I'm not sure how it's being done now, but I just wanted to tell you, that when some string is changed in English (correction or new features) it is very much wanted that it doesn't go live immediately, but instead it's changed on beta (for everyone involved to see the changes in action, but don't ruin the feel of a fully translated interface for ordinary users), uploaded to transifex, and only after, say, 2-3 days the changes take effect simultaneously in English and other languages.

**Comment by paglias** [comment]
@GitHubSphinx enough time, sonthing is still broken (drops, again.......) and I'll leave till sunday so enough time and probably some days more but no updates from Transifex in the meantime unless someone else updates the files and I don't think so. Some things are necessarely english only (quest messages in the chat, for example). Waiting for translations: a good idea, cc @lefnire

**Comment by ghost** [comment]
@paglias Thank you! As users see the translation they come up with suggestions to improve some strings. These suggestions are not critical, but some of them may be considered to improve user experience. E.g. with a new combined toolbar "Site" in Options becomes not evident at all, why not change it into something more compatible with "Options" - for example, "General". Or another thing, "Language" in the settings could be changed to "Interface Language" to avoid possible confusion. They suggest tailoring translated strings, but I'm convinced it may not be the right solution and it's better to improve original English strings to avoid disconnection between English and translated files.

@paglias @deilann 
So what do you think about this? Can we have some place to gather such suggestions of improving text strings all together like in case with typos? Or should I try to meet the demands of users by tailoring only the translated strings?
